### PR TITLE
fix(dev-overlay): align codeframe to formatIssue

### PR DESCRIPTION
**Pull request by kwonoj**
Closes PACK-2825

### CODE REVIEW

**Comment by ijjk** [comment]
## Failing test suites 
Commit: a7bdeeb8fd6c9a2ac006f5bb87d29ab0a1292dce

`pnpm test test/integration/config-devtool-dev/test/index.test.js`
- devtool set in development mode in next config > should warn and revert when a devtool is set in development mode

<details>
<summary>Expand output</summary>

  ● devtool set in development mode in next config › should warn and revert when a devtool is set in development mode

    expect(received).toMatchInlineSnapshot(snapshot)

    Snapshot name: `devtool set in development mode in next config should warn and revert when a devtool is set in development mode 1`

    - Snapshot  - 4
    + Received  + 4

      pages/index.js (5:11) @ eval

    -   3 | export default function Index(props) {
        4 |   useEffect(() => {
    - > 5 |     throw new Error('this should render')
    +   5 |     throw new Error('this should render')
    -     |           ^
    -   6 |   }, [])
    + > 6 |   }, [])
    +     |            ^
        7 |   return <div>Index Page</div>
        8 | }
    +   9 |

      40 |         // TODO: add win32 snapshot
      41 |       } else {
    > 42 |         expect(await getRedboxSource(browser)).toMatchInlineSnapshot(`
         |                                                ^
      43 |           "pages/index.js (5:11) @ eval
      44 |
      45 |             3 | export default function Index(props) {

      at Object.toMatchInlineSnapshot (integration/config-devtool-dev/test/index.test.js:42:48)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

**Comment by ijjk** [comment]
# Stats from current PR

<details open>
<summary><strong>Default Build</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | vercel/next.js source-issue-imports-error | Change |
| - | - | - | - |
| buildDuration | 13.8s | 13.7s | N/A |
| buildDurationCached | 8.4s | 6.1s | N/A |
| nodeModulesSize | 198 MB | 198 MB | N/A |
| nextStartRea..uration (ms) | 430ms | 436ms | N/A |

</details>

<details>
<summary><strong>Client Bundles (main, webpack)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js source-issue-imports-error | Change |
| - | - | - | - |
| 2453-HASH.js gzip | 30.8 kB | 30.8 kB | N/A |
| 3304.HASH.js gzip | 181 B | 181 B | ✓ |
| 3f784ff6-HASH.js gzip | 53.7 kB | 53.7 kB | ✓ |
| 8299-HASH.js gzip | 5.04 kB | 5.04 kB | N/A |
| framework-HASH.js gzip | 45.2 kB | 45.2 kB | ✓ |
| main-app-HASH.js gzip | 243 B | 242 B | N/A |
| main-HASH.js gzip | 32.2 kB | 32.2 kB | N/A |
| webpack-HASH.js gzip | 1.68 kB | 1.68 kB | N/A |
| Overall change | 99 kB | 99 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js source-issue-imports-error | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | vercel/next.js source-issue-imports-error | Change |
| - | - | - | - |
| _app-HASH.js gzip | 196 B | 197 B | N/A |
| _error-HASH.js gzip | 184 B | 184 B | ✓ |
| amp-HASH.js gzip | 505 B | 505 B | ✓ |
| css-HASH.js gzip | 324 B | 325 B | N/A |
| dynamic-HASH.js gzip | 2.5 kB | 2.5 kB | N/A |
| edge-ssr-HASH.js gzip | 258 B | 258 B | ✓ |
| head-HASH.js gzip | 352 B | 352 B | ✓ |
| hooks-HASH.js gzip | 370 B | 371 B | N/A |
| image-HASH.js gzip | 4.21 kB | 4.21 kB | ✓ |
| index-HASH.js gzip | 259 B | 259 B | ✓ |
| link-HASH.js gzip | 2.67 kB | 2.67 kB | N/A |
| routerDirect..HASH.js gzip | 314 B | 312 B | N/A |
| script-HASH.js gzip | 386 B | 386 B | ✓ |
| withRouter-HASH.js gzip | 309 B | 309 B | ✓ |
| 1afbb74e6ecf..834.css gzip | 106 B | 106 B | ✓ |
| Overall change | 6.57 kB | 6.57 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | vercel/next.js source-issue-imports-error | Change |
| - | - | - | - |
| _buildManifest.js gzip | 481 B | 484 B | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js source-issue-imports-error | Change |
| - | - | - | - |
| index.html gzip | 528 B | 527 B | N/A |
| link.html gzip | 541 B | 539 B | N/A |
| withRouter.html gzip | 524 B | 519 B | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Edge SSR bundle Size</strong></summary>

|  | vercel/next.js canary  | vercel/next.js source-issue-imports-error | Change |
| - | - | - | - |
| edge-ssr.js gzip | 95.3 kB | 95.3 kB | N/A |
| page.js gzip | 3.04 kB | 3.04 kB | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Middleware size</strong></summary>

|  | vercel/next.js canary  | vercel/next.js source-issue-imports-error | Change |
| - | - | - | - |
| middleware-b..fest.js gzip | 625 B | 622 B | N/A |
| middleware-r..fest.js gzip | 151 B | 151 B | ✓ |
| middleware.js gzip | 25.5 kB | 25.5 kB | N/A |
| edge-runtime..pack.js gzip | 839 B | 839 B | ✓ |
| Overall change | 990 B | 990 B | ✓ |

</details>

<details>
<summary><strong>Next Runtimes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js source-issue-imports-error | Change |
| - | - | - | - |
| app-page-exp...dev.js gzip | 170 kB | 170 kB | ✓ |
| app-page-exp..prod.js gzip | 96.8 kB | 96.8 kB | ✓ |
| app-page-tur..prod.js gzip | 98.5 kB | 98.5 kB | ✓ |
| app-page-tur..prod.js gzip | 92.8 kB | 92.8 kB | ✓ |
| app-page.run...dev.js gzip | 144 kB | 144 kB | ✓ |
| app-page.run..prod.js gzip | 91.3 kB | 91.3 kB | ✓ |
| app-route-ex...dev.js gzip | 21.4 kB | 21.4 kB | ✓ |
| app-route-ex..prod.js gzip | 15.1 kB | 15.1 kB | ✓ |
| app-route-tu..prod.js gzip | 15.1 kB | 15.1 kB | ✓ |
| app-route-tu..prod.js gzip | 14.8 kB | 14.8 kB | ✓ |
| app-route.ru...dev.js gzip | 21 kB | 21 kB | ✓ |
| app-route.ru..prod.js gzip | 14.8 kB | 14.8 kB | ✓ |
| pages-api-tu..prod.js gzip | 9.55 kB | 9.55 kB | ✓ |
| pages-api.ru...dev.js gzip | 9.82 kB | 9.82 kB | ✓ |
| pages-api.ru..prod.js gzip | 9.55 kB | 9.55 kB | ✓ |
| pages-turbo...prod.js gzip | 22.5 kB | 22.5 kB | ✓ |
| pages.runtim...dev.js gzip | 23.1 kB | 23.1 kB | ✓ |
| pages.runtim..prod.js gzip | 22.4 kB | 22.4 kB | ✓ |
| server.runti..prod.js gzip | 50.9 kB | 50.9 kB | ✓ |
| Overall change | 943 kB | 943 kB | ✓ |

</details>

<details>
<summary><strong>build cache</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | vercel/next.js source-issue-imports-error | Change |
| - | - | - | - |
| 0.pack gzip | 1.57 MB | 1.57 MB | ⚠️ +2.15 kB |
| index.pack gzip | 106 kB | 106 kB | N/A |
| Overall change | 1.57 MB | 1.57 MB | ⚠️ +2.15 kB |

</details>

<details>
<summary><strong>Diff details</strong></summary>

<details>
<summary>Diff for <strong>middleware.js</strong></summary>

Diff too large to display
</details>

</details>

</details>
Commit: a7bdeeb8fd6c9a2ac006f5bb87d29ab0a1292dce

File: test/development/acceptance-app/ReactRefreshLogBox-builtins.test.ts [hunk 1]
```diff
@@ -58,6 +58,8 @@ describe.each(['default', 'turbo'])('ReactRefreshLogBox app %s', () => {
             |             ^^^^^^^^^^^^^^
           2 | module.exports = dns
 
+        Import map: No import map entry
```
Comment (wbinnssmith) [inline test/development/acceptance-app/ReactRefreshLogBox-builtins.test.ts:61]: I don't think we should show this to users

Comment (kwonoj) [reply test/development/acceptance-app/ReactRefreshLogBox-builtins.test.ts:61]: I agree, but it just appears somehow?

File: packages/next/src/server/dev/turbopack-utils.ts [hunk 2]
```diff
@@ -118,7 +118,14 @@ export function formatIssue(issue: Issue) {
   }
 
   if (description) {
-    message += renderStyledStringToErrorAnsi(description) + '\n\n'
+    if (
+      description.type === 'text' &&
+      description.value.startsWith('Import map: No import map entry')
```
Comment (sokra) [inline packages/next/src/server/dev/turbopack-utils.ts:123]: Let's remove this message from turbopack instead, only showing it if there is a sourcemap

Comment (kwonoj) [reply packages/next/src/server/dev/turbopack-utils.ts:123]: Where do we emit this message? I could see where it comes from (https://github.com/vercel/turbo/blob/main/crates/turbopack-core/src/resolve/options.rs#L332) and some existing test have same (https://github.com/vercel/turbo/blob/97950cd84b7278b6b3b16ee9bc9d96c303ddd4d6/crates/turbopack-tests/tests/snapshot/imports/resolve_error_esm/issues/__l___Module%20not%20found____c__%20Can't%20resolve%20'__c_d-b93f4e.txt#L11) but unsure where it is being emitted.

**Review by wbinnssmith — APPROVED** [review]

**Pull request merged**